packages/pools/src/concentrated/fetch-pool-amount-provider.ts

```diff
@@ -9,19 +9,36 @@ type BankAmountResponse = {
   }[];
 };
 
-/** Provider of bank balances from bank module by bech32Address. Expected by concentrated liquidity pool. */
+/** Provider of bank balances from bank module by bech32Address. Expected by concentrated liquidity pool.
+ *  The instance lifecycle is assumed to follow the pool instance.
+ *  Maintains a static class, so multiple instances of this class will share the same cache per query node.
+ */
 export class FetchPoolAmountDataProvider implements AmountsDataProvider {
   /** Address => Response */
-  protected amountsCache = new Map<
+  protected static _amountsCache = new Map<
     string,
     Awaited<ReturnType<AmountsDataProvider["getPoolAmounts"]>>
   >();
 
-  protected activeCacheTimeouts = new Map<
+  /** Address => Timeout ID */
+  protected static _activeCacheTimeouts = new Map<
     string,
     ReturnType<typeof setTimeout>
   >();
 
+  /** Map of `bech32Address: string` => Promise if fetching for that address. */
+  protected static _inFlightFetchPerAccount = new Map<
```

Comment:
What do you think if we make the name more generic since pool addresses could also be used here? 

```suggestion
  protected static _inFlightFetchPerAddress = new Map<
```